fetch.bs

```diff
@@ -8584,10 +8593,67 @@ method steps are:
         https://github.com/whatwg/dom/issues/1031#issuecomment-1233206400 -->
   </ol>
 
+ <li><p>Let <var>hasUploadListeners</var> be false.
+
+ <li><p>Let <var>requestObserver</var> be null.
+
+ <li><p>Let <var>responseObserver</var> be null.
+
+ <li>
+  <p>If <var>init</var>["{{RequestInit/observer}}"] <a for=map>exists</a>, then:
+
+  <ol>
+   <li><p>Let <var>observerCallback</var> be <var>init</var>["{{RequestInit/observer}}"].
+
+   <li><p>Set <var>requestObserver</var> to a {{FetchObserver}}.
+
+   <li><p>Set <var>responseObserver</var> to a {{FetchObserver}}.
+
+   <li><p>Let <var>args</var> be « <var>requestObserver</var>, <var>responseObserver</var> ».
+
+   <li><p>[=invoke|Invoke=] <var>observerCallback</var> with <var>args</var>
+   and <code>"rethrow"</code>. If this throws an exception, <a for=/>reject</a> <var>p</var> with it
+   and return <var>p</var>.
+
+   <li><p>If one or more <a event><code>progress</code></a> event listeners were added to
+   <var>requestObserver</var>, then set <var>hasUploadListeners</var> to true.
+  </ol>
+
+ <li><p>Let <var>requestBodyTransmitted</var> be 0.
+
+ <li><p>Let <var>requestBodyLength</var> be <var>request</var>'s <a for=request>body</a>'s
+ <a for=body>length</a>, if <var>request</var>'s <a for=request>body</a> is non-null;
+ otherwise 0.
+
+ <li><p>Assert: <var>requestBodyLength</var> is an integer.
+
+ <li>
+  <p>Let <var>processRequestBodyChunkLength</var>, given a <var>bytesLength</var>, be these steps:
+
+  <ol>
+   <li><p>Increase <var>requestBodyTransmitted</var> by <var>bytesLength</var>.
+
+   <li><p>If not roughly 50ms has passed since these steps were last invoked, then return.
+
+   <li><p>If <var>hasUploadListeners</var> is true, then <a>fire a progress event</a> named
+   <a event><code>progress</code></a> at <var>requestObserver</var> with <var>requestBodyTransmitted</var>
+   and <var>requestBodyLength</var>.
+  </ol>
+
  <li>
-  <p><p>Set <var>controller</var> to the result of calling <a for=/>fetch</a> given
-  <var>request</var> and <a for=fetch><i>processResponse</i></a> given <var>response</var> being
-  these steps:
+  <p>Let <var>processRequestEndOfBody</var> be these steps:
+
+  <ol>
+   <li><p>If <var>hasUploadListeners</var> is false, then return.
+
+   <li><p>Increase <var>requestBodyTransmitted</var> by <var>bytesLength</var>.
+
+   <li><p><a>Fire a progress event</a> named <a event><code>progress</code></a> at <var>requestObserver</var>
+   with <var>requestBodyTransmitted</var> and <var>requestBodyLength</var>.
+  </ol>
+
+ <li>
+  <p>Let <var>processResponse</var> given a <var>response</var> be these steps:
```

Comment:
If we want to do progress tracking for response we'll need to do it differently to XHR, XHR consumes the body inside of this algorithm but fetch obviously can't.

One idea is to wrap response's body with a wrapper readable stream that allows use to fire progress events when the body actually gets read. This way we don't consume anything to report progress, (we can make it so we only wrapper if there's a responseObserver with a progress event listeners).

Reply:
Why couldn't the event be fired here https://whatpr.org/fetch/1843.html#http-network-fetch:~:text=Increase%20response%E2%80%99s%20body%20info%E2%80%99s%20encoded%20size%20by%20bytes%E2%80%99s%20length%2E ?